.github/actions/install_toolchain/action.yml

```diff
@@ -0,0 +1,16 @@
+name: "Install toolchains"
+description: "Install toolchains"
```

Comment:
I feel that if we're going to refactor our github actions, we should consider using Nix like in our local E2E, too avoid duplicating setup routines.

* https://github.com/marketplace/actions/install-nix#usage-with-flakes
* https://github.com/marketplace/actions/install-nix
*

Reply:
Agree!  Current refactor is mainly for extracting a common rust toolchain to avoid duplicated declarations. For the whole Nix setup refactor could we do in another PR?